### PR TITLE
make garbage collection interval configurable

### PR DESCRIPTION
**Pull request by epwalsh**

### CODE REVIEW

File: olmo/config.py [hunk 1]
```diff
@@ -987,6 +987,11 @@ class TrainConfig(BaseConfig):
     How often to log to the console.
     """
 
+    gen1_gc_interval: int = 1
```
Comment (2015aroras) [inline olmo/config.py:990]: nit: it might be nice to provide an easy way to turn this off (e.g. setting it to None)

**Review by 2015aroras — APPROVED** [review]

File: olmo/config.py (hunk 1 above)
Comment (epwalsh) [reply olmo/config.py]: Done! 989f7992

(I assumed you mean disabling manual collection, i.e. not messing with it at all)

**Review by dirkgr — APPROVED** [review]

File: olmo/config.py [hunk 2]
```diff
@@ -987,9 +987,10 @@ class TrainConfig(BaseConfig):
     How often to log to the console.
     """
 
-    gen1_gc_interval: int = 1
+    gen1_gc_interval: Optional[int] = 1
```
Comment (2015aroras) [inline olmo/config.py:990]: Just checking, we want the default behavior to be messing with GC?

**Review by 2015aroras — APPROVED** [review]

File: olmo/config.py (hunk 2 above)
Comment (epwalsh) [reply olmo/config.py:990]: Yes I think so. There might be a better default interval for smaller models, but this was a huge improvement for the 70B.

**Pull request merged**